最終課題3.テーブル設計

unitoro0903 opened the pull request:
https://gyazo.com/22fcd5aef661283d789b6a4168d8f831

unitoro0903 committed:
最終課題3.テーブル設計

yuri240 commented on README.md:
```diff
@@ -22,3 +22,56 @@ Things you may want to cover:
+
+## users table
+
+| Column             | Type                | Options                   |
```
ユーザーテーブルに必要なカラムが不足しているようです。

見本アプリの**新規登録画面**を確認し、
入力フォームの数分、カラムを追加しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,56 @@ Things you may want to cover:
+
+| Column                              | Type       | Options                        |
+|-------------------------------------|------------|--------------------------------|
+| title                               | string     | null: false                    |
```
商品テーブルに必要なカラムが不足しているようです。

見本アプリの**商品出品画面**を確認し、
入力フォームの数分、カラムを追加しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,56 @@ Things you may want to cover:
+
+| Column      | Type       | Options                        |
+|-------------|------------|--------------------------------|
+| address     | text       | null: false                    |
```
住所テーブルに必要なカラムが不足しているようです。

見本アプリの**商品購入画面**を確認し、
配送先住所に関する入力フォームの数分、カラムを追加しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,56 @@ Things you may want to cover:
+### Association
+
+- belongs_to :items
+- belongs_to :users
```
`belongs_to`は単数形のアソシエーションとなりますので、単語の末尾は全て**単数形**にしましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。
また、次回ご提出の際にはER図も修正が必要となります。

お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。

unitoro0903 committed:
修正１

https://gyazo.com/757083ea906a3444663288957a4ac3d9

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| Column             | Type                | Options                   |
+|--------------------|---------------------|---------------------------|
+| email              | string              | null: false, unique: true |
+| password           | string              | null: false               |
```
```
| password           | string              | null: false               |
```
こちらのカラムはdeviseのgemを使用しての実装になるため、deviseが予め用意している
`encrypted_password`というカラム名に変更しましょう。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| password           | string              | null: false               |
+| name               | string              | null: false               |
+| nickname           | string              | null: false               |
+| birthday           | string              | null: false               |
```
```
| birthday           | string              | null: false               |
```
誕生日のカラムは「date型」を用いることで年、月、日を1つにして保存できます。

integerからdateに変更しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+|--------------------|---------------------|---------------------------|
+| email              | string              | null: false, unique: true |
+| password           | string              | null: false               |
+| name               | string              | null: false               |
```
名前に関するカラムが不足しているようです。

ユーザー新規登録画面を確認し、入力フォームの数分、カラムを用意しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| name               | string              | null: false               |
+| nickname           | string              | null: false               |
+| birthday           | string              | null: false               |
+| address            | text                | null: false               |
```
ユーザー新規登録画面にアドレス入力欄はないかと存じます。
したがって、こちらのカラムは削除しておきましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| Column                              | Type       | Options                        |
+|-------------------------------------|------------|--------------------------------|
+| title                               | string     | null: false                    |
+| price                               | string     | null: false                    |
```
priceカラムは数値のみを登録するカラムなため、integer型で設計しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| title                               | string     | null: false                    |
+| price                               | string     | null: false                    |
+| user(出品者)                        | references  | null: false, foreign_key: true |
+| image                               | text       | null: false                    |
```
imageカラムですが、

今回の実装では商品出品の画像投稿において「active_storage」を使用するため、
設計の段階から削除しておきましょう。

理由：active_storage導入時に自動でテーブルなどが生成されるからです。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| image                               | text       | null: false                    |
+| explanation                         | string     | null: false                    |
+| category                            | string     | null: false                    |
+| condition                           | string     | null: false                    |
```
**商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリー**のカラムは
active_hashで実装の予定のため、カラム名には`「_id」`をつけ、`integer`型にしましょう！

詳しくは「`最終課題補足カリキュラム 〜カテゴリーの選択を工夫しよう〜`」のカリキュラムを参考にしていただけると幸いです。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+
+| Column      | Type       | Options                        |
+|-------------|------------|--------------------------------|
+| region      | text       | null: false                    |
```
```
| region      | text       | null: false                    |
```
こちらの都道府県カラムは、
active_hashのデータとして使い回すことができますので、
itemsテーブルの都道府県カラムと同じものを使用しましょう。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+| Column      | Type       | Options                        |
+|-------------|------------|--------------------------------|
+| region      | text       | null: false                    |
+| item(商品のID)　| references | null: false, foreign_key: true |
```
配送先住所テーブルに商品IDを関連付ける必要はございません。
８０行目のカラムは削除しておきましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,68 @@ Things you may want to cover:
+|-------------|------------|--------------------------------|
+| region      | text       | null: false                    |
+| item(商品のID)　| references | null: false, foreign_key: true |
+| record(購入情報のid) | references | null: false, foreign_key: true |
```
８１行目のカラムはrecord（`(購入情報のid) `という日本語の説明書きは削除する）としておきましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。


また、**次回ご提出の際にはER図も修正が必要**となります。

**お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。**

unitoro0903 committed:
修正２

「 region      | text       | null: false
こちらの都道府県カラムは、
active_hashのデータとして使い回すことができますので、
itemsテーブルの都道府県カラムと同じものを使用しましょう。」 の部分がわからなくて、itemsテーブルに都道府県カラムを移すのでしょうか？itemsテーブルに都道府県は必要でしょうか？

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+| email              | string              | null: false, unique: true |
+| encrypted_password | string              | null: false               |
+| kanjiname          | string              | null: false               |
+| kananame           | string              | null: false               |
```
名前に関するカラムは、
`苗字・名前・苗字（フリガナ）・名前（フリガナ）`と合計4つにしましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+| user(出品者)                        | references  | null: false, foreign_key: true |
+| explanation                         | string     | null: false                    |
+| category.id                         | integer     | null: false                    |
+| condition.id                        | integer     | null: false                    |
```
アクティブハッシュのカラムは合計5つ用意しましょう！

また、カラム名の末尾は`⚪︎⚪︎⚪︎_id`としていただく必要がございます。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+### Association
+
+- belongs_to :user
+- belongs_to :record
```
`- belongs_to :record`

６０行目のアソシエーションは、1対1を示す`has_one`メソッドを使用しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+### Association
+
+- belongs_to :item
+- belongs_to :user
```
recordsテーブルにshipモデルのアソシエーションも設定しましょう！

理由：
配送先住所を登録する際、購入記録も同時登録するため、モデル同士の関連付けが必要となります。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+
+| Column      | Type       | Options                        |
+|-------------|------------|--------------------------------|
+| region.id   | integer    | null: false                    |
```
７８行目

こちらもアクティブハッシュのカラムとなりますので、`region_id`にしましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+| region.id   | integer    | null: false                    |
+| record      | references | null: false, foreign_key: true |
+| shipfee     | string     | null: false                    |
+| recuiretime | string     | null: false                    |
```
配送先住所テーブルのカラムが不足しているようです。

商品購入ページを確認し、配送先住所の入力フォーム分、カラムを用意しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,66 @@ Things you may want to cover:
+
+### Association
+
+- belongs_to :item
```
shipsテーブルにitemを紐づける必要はございませんので、８７行目は削除しておきましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。

追加でコメントさせていただきましたので、ご確認お願いします。
また、ご不明な点がありましたら、コネクトメンターへご質問いただきますようお願い致します。

### **次回ご提出の際にはER図も修正が必要**となります。

**お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。**

unitoro0903 committed:
修正３

unitoro0903 commented on README.md:
https://gyazo.com/1eceffcf8befb762dd7048644013bdfd

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+| kanjifirstname     | string              | null: false               |
+| kanjifamilyname    | string              | null: false               |
+| kanafirstname      | string              | null: false               |
+| kanafamilyname     | string              | null: false               |
```
2つ以上の単語を繋ぐ場合は、アンダースコアを使用しましょう！

例）`first_name`,`family_name`,`first_name_kana`など

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+|--------------------|-------------------|------------------------|
+| title              | string            | null: false            |
+| price              | integer           | null: false            |
+| user.id            | references | null: false, foreign_key: true |
```
52,54,55行目のカラムについて

カラム名に「`.(ドット`)」を用いることはできないため、
「`_(アンダースコア)`」を使用しましょう！

例）`category_id`

また、**references**型を使用した場合、カラム名末尾の`_id`は省略可能となりますので、52行目については、userとしましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+| user.id            | references | null: false, foreign_key: true |
+| explanation        | string     | null: false                   |
+| category.id        | integer           | null: false              |
+| condition.id      | integer           |null:false                 |
```
プルダウン選択のカラムが不足しているため、商品出品画面を確認し、カラムを追加しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+
+| Column      | Type       | Options                        |
+|-------------|------------|--------------------------------|
+| region.id   | integer    | null: false                    |
```
82行目のカラムについてもドットは使用できないため、`region_id`としましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+| record      | references | null: false, foreign_key: true |
+| shipfee.id  | string     | null: false                    |
+| recuiretime.id | string     | null: false                 |
+| shikutyouson  | string     | null:false                     |
```
`shikutyouson`、`banchi`

カラム名は、ローマ字表記ではなく、英単語にしましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+| shikutyouson  | string     | null:false                     |
+| banchi      | string     | null:false                     |
+| phonenumber | string     | null:false                     |
+| postcode    | string     | null:false                     |
```
2つ以上の単語を繋ぐ場合は、アンダースコアを使用しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,72 @@ Things you may want to cover:
+| region.id   | integer    | null: false                    |
+| record      | references | null: false, foreign_key: true |
+| shipfee.id  | string     | null: false                    |
+| recuiretime.id | string     | null: false                 |
```
84,85行目のカラムは文字列を登録するカラムとなりますので、
カラム名に`id`を使用する必要はないかと存じます。

補足：第三者がカラム名を見て、何を登録するカラムかがわかるよう命名していただけますと良さそうです。

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
キャプチャ画像と修正点確認させていただきました。

追加でコメントさせていただきましたので、ご確認お願いします。

また、次回ご提出の際にはER図も修正が必要となります。

お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。

unitoro0903 committed:
修正４

unitoro0903 commented on README.md:
https://gyazo.com/f0caad05123954a4744c8bf46a671df1

TakashiKozuma commented on README.md:
```diff
@@ -22,3 +22,74 @@ Things you may want to cover:
+| street_address      | string     | null:false             |
+| phone_number | string     | null:false                     |
+| postcode    | string     | null:false                     |
+| building_name      | string     | null:false             |
```
```Rb
| building_name | string | null:false |
```
建物名は必須ではないので、null:falseは削除しましょう。

TakashiKozuma reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
コメントを追記致しましたので、ご確認お願いします。
また、次回レビュー依頼時に最新状態のER図の添付もお願いいたします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

unitoro0903 committed:
修正5

unitoro0903 commented on furima.dio:
https://gyazo.com/40c4e3e47400d602eebf9c87b875c17d

unitoro0903 committed:
修正6

https://gyazo.com/1b0cf097148bbcfb8ce3a13abaf081b0

unitoro0903 commented on furima.dio:
こちら間違えて投稿しました。修正6をご確認お願いいたします。

minachanchan commented on README.md:
```diff
@@ -22,3 +22,74 @@ Things you may want to cover:
+| title              | string            | null: false            |
+| price              | integer           | null: false            |
+| user            | references | null: false, foreign_key: true |
+| explanation        | string     | null: false                   |
```
```ruby
| explanation | string | null: false |
```
商品説明カラムのデータ型はtext型を使用しましょう！
string型ですと、文字列の個数が255字までと制限されてしまうためです。

wad-sho replied:
`| explanation        | string     | null: false                   |`

商品説明用カラムは長文を許可したいためtext型に修正しましょう！

minachanchan reviewed (commented):
実装お疲れ様です！レビューのご依頼ありがとうございます。

コメントいたしましたので、ご確認頂けますと幸いでございます。

指摘内容の修正がお済みになりましたら、再度レビュー依頼くださいませ。
その際、ER図もREADMEの内容と合わせて修正とお願いいたします。
**現在READMEと完全に一致していないものが散見されますので、テーブル名、カラム名全て一致するよう修正をお願いいたします！**
**次回のレビュー依頼の際に最新のER図のURLをまたこちらのプルリクエストに添付頂ければと存じます。**

実装完了まであと少し、一緒に頑張りましょう！
何卒よろしくお願いいたします。

wad-sho reviewed (commented):
レビュー依頼ありがとうございます。
コメントいたしましたので、ご確認お願いします。

指摘内容の修正が終わったら、再度レビュー依頼ください。
その際、ER図もREADMEの内容と合わせて修正とお願いいたします。

**次回のレビュー依頼の際に最新のER図のURLをまたこちらのプルリクエストに添付お願いします**

unitoro0903 committed:
修正7

https://gyazo.com/c16ae2fb7a3bf48740a8335d7ecfe555

wad-sho reviewed (commented):
レビュー依頼ありがとうございます！
実装内容を確認しました。

綺麗にまとまっております。

**データベース設計、LGTMとさせていただきます。実装お疲れ様でした！**

このブランチをマージし、masterブランチ（もしくはmainブランチ）でプルしましょう。
内容が反映されていることを確認し、次の実装・学習に入りましょう！
引き続き応援しております！
